tests/virt_autotest/login_console.pm

```diff
@@ -67,7 +67,7 @@ sub login_to_console {
         ipmitool("chassis power reset");
         reset_consoles;
         select_console 'sol', await_console => 0;
-        assert_screen([qw(grub2 grub1 prague-pxe-menu)], 90);
+        check_screen([qw(grub2 grub1 prague-pxe-menu)], 90);
     }
```

Comment:
Although this can has some higher chance to continue test when ipmi sol console behaves abnormally/unstablely, it is still hard to have it successfully continue every time.
Our tests are run on ipmi backend also. So this is also what we are suffering from. Instead of enhancing the sol console related check points one by one, we choose another way which is the discussion result with coolo -- a job retrigger tool to automatically rerun jobs fail at sol console related steps to overcome ipmi unstability. I cc you the email introducing it. Maybe you can enjoy it as a supplement to avoid manual review and retrigger when jobs fail not by product but by ipmi itself unstable behaviors.

Reply:
> it is still hard to have it successfully continue every time.
Can you elaborate on this? From my experience a lot of tests are failing here because of hardware issues which we cannot avoid. I think it is safe to let the test pass even if no GRUB is shown on the SOL console.

Reply:
When you say hardware issue, I guess it is equivalent to ipmi board sol console unstability. There can be various different wrong behaviors  when it is not working well, eg no sol console screen update or black screen, wrong typing  with simulated board input, not responding to user interaction or very slow responding. So when ipmi sol console does not respond/update any more from this grub point, there can be no way to let it pass with next assert_screen/type_string with just letting it go to next step. All in all, it is hard to get 100% success from test/backend code when ipmi facility itself is not 100% behaving well, especially when it stops functioning from some time point until next "mc reset".

Reply:
I think there is a way. When the IPMI stops functioning the test normally dies because of this line `assert_screen` but if we use `check_screen` the test can continue and at the end it will simply switch to SSH and because of the system had enough time to boot up we can simply continue.

Reply:
Yes, for this specific step, it can be a way. I just noticed you changed all assert_screen to check_screen in this file, and with this if you no longer has any interactions like send_key/type_string on sol console in your tests either in this file or later test pm files, it can be fine . But what about the following send_key/type_string requests in general?

Reply:
It will either go through or fail as it would fail anyways, just a bit later. But it will still be visible where the error goes from (the reviewer will see black screen instead of grub). In my opinion this does not make the situation worse nor more chaotic - this just make the situation time to time better :)

Reply:
Yes, agree. Please take more comment more as hints, rather than rejection to this change. It does improves on some situations.